### ISSUE TITLE
orgmode: follow link (SPC m l or evil-org-open-links) does not work

### ISSUE THREAD

**Issue by sotte** [syl20bnr/spacemacs#1429]
Using evil and orgmode `SPC m l` and `, l` should open links, but it does not. Clicking on the link open a new buffer though.

This would be a minimal org file
```org
[[file:my_file.org]]
```

**Comment by wsdookadr** [comment]
Is this [`SPC m l`](https://github.com/syl20bnr/spacemacs/blob/8f75ce07b2797ee67c56d3011c25cedb8a8783f4/contrib/org/README.md#org-with-evil-org-mode) which does [`evil-org-open-links`](https://github.com/syl20bnr/spacemacs/blob/8f75ce07b2797ee67c56d3011c25cedb8a8783f4/contrib/org/packages.el#L43) . Recently, a bug was solved in relation to that here [evil-org-mode/issue/14](https://github.com/edwtjo/evil-org-mode/issues/14) .

**Comment by sotte** [comment]
Just a typo, it should have been `SPC m l` (I corrected it in the first post). I'm nor sure how related my problem is. Emacs does not crash if I want to open a link.

Just summarzing
- `SPC m l`, `, l`, and `evil-org-open-links` don't not open the link, nothing happens
- `org-open-at-point` works (that's what I'm using as a workaround right now)
- `ace-link-org` also opens links
- clicking on the link works

**Comment by wsdookadr** [comment]
@sotte `evil-org-open-links` (and `SPC m l`) opens the links in a visual selection. is this how you're using it ?

**Comment by sotte** [comment]
My understanding is that `evil-org-open-links` opens the link under the cursor. No visual selection necessary.

I just tested it with visual selection and then I actually get the infinite look as described in https://github.com/edwtjo/evil-org-mode/issues/14.

**Comment by wsdookadr** [comment]
@sotte can you try the patch in that pull request ? it should fix the infinite loop.

**Comment by rakyi** [comment]
`SPC m o` is by default mapped to `evil-org-open-links` now, but does nothing as far as I can tell. However `org-open-at-point` opens links as expected. Should we map `SPC m o` to it instead? I'm new to Org mode and I don't know what  is `evil-org-open-links` supposed to do.

Also clicking on link with mouse works, but makes a visual selection from the link clicked to its target, when it is pointing to another Org node.

**Comment by syl20bnr** [comment]
`SPC m o` is now reserved to the user, in develop it is ` SPC m l`. We can use the regular org function instead.

**Comment by TheBB** [comment]
I had a go with this and `evil-org-open-links` does work, but it's a bit… weird.

It opens all links inside the visual selection, so if you don't have a visual selection it will do nothing. It also requires the *start* of the link to be inside the selection, instead of just any part of it. Also, since some links have hidden characters in Org (i.e. the `[[` and `]]`), you can't just select the first part of the address, typically you have to select the character before the link as well.

E.g. if the buffer contents are `[[http://www.google.com]]`, they are displayed as `http://www.google.com`, but if you select the `http`, `evil-org-open-links` will not find anything. What you have to do is add a space before, so that the buffer contents are ` [[http://www.google.com]]`, which is displayed as ` http://www.google.com`, so that you can select ` http` (with the space before it). Then, `evil-org-open-links` will open it.

In summary, it's pretty hairy to get it to do what you want. I support just binding `org-open-at-point` to `,l`.

**Comment by TheBB** [comment]
Closed by 194f7d5.

**Comment by TheBB** [comment]
Ah, this is in master. So we'll keep this open until release.
